### PR TITLE
Add support for retrieval of dependent types and associations

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Related issue: https://github.com/ballerina-platform/ballerina-standard-library/issues/4087

## Checklist
- [x] Linked to an issue
- [x] Updated the specification
- [ ] Updated the changelog
- [x] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/sql_client.bal [hunk 1]
```diff
@@ -141,6 +159,46 @@ public client class SQLClient {
         }
     }
 
+        # Retrieves the values of the 'many' side of an association.
```
Comment (kalaiyarasiganeshalingam) [inline ballerina/sql_client.bal:162]:
```suggestion
    # Retrieves the values of the 'many' side of an association.
```

File: ballerina/metadata_types.bal [hunk 2]
```diff
@@ -22,21 +22,80 @@ import ballerina/sql;
...
+# + ONE_TO_ONE - The association type is a one-to-one association
+# + ONE_TO_MANY - The entity is in the 'one' side of a one-to-many association
+# + MANY_TO_ONE - The entity is in the 'many' side of a one-to-many association
+public enum JoinType {
+    ONE_TO_ONE,
+    ONE_TO_MANY,
+    MANY_TO_ONE
+}
+
+//regex for "type" keyword following by string of characters
```
Comment (daneshk) [inline ballerina/metadata_types.bal:101]:
```suggestion
```

File: ballerina/tests/associations-tests.bal [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+            }
+        };
+
+    if !found {
+        test:assertFail("Expected BuildingInfo not found.");
+    }
+
+    check rainierClient.close();
+
+}
```
Comment (daneshk) [inline ballerina/tests/associations-tests.bal:293]:
```suggestion
}

```

File: ballerina/types.bal [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import ballerina/sql;
```
Comment (daneshk) [inline ballerina/types.bal:1]: Shall we add license headers

File: examples/medical-center/Ballerina.toml [hunk 5]
```diff
@@ -3,15 +3,21 @@ org = "wso2"
...
-name = "persist"
-version = "0.1.0"
-repository = "local"
-
 [persist.medical_center.storage.mysql]
 host = "localhost"
 port = 3305
 user = "root"
 password = "Test123#"
 database = "medical_center_db"
```
Comment (daneshk) [inline examples/medical-center/Ballerina.toml:11]:
```suggestion
[persist]
datastore = "mysql"
module = "medical_center"
```

Comment (kaneeldias) [reply examples/medical-center/Ballerina.toml]: Fixed in [3ac4df4](https://github.com/ballerina-platform/module-ballerina-persist/pull/155/commits/3ac4df4447e4c445dda487eb932865f44b7f670c)

File: ballerina/types.bal (hunk 4 above)
Comment (kaneeldias) [reply ballerina/types.bal:16]: Fixed in [3ac4df4](https://github.com/ballerina-platform/module-ballerina-persist/pull/155/commits/3ac4df4447e4c445dda487eb932865f44b7f670c)

File: ballerina/tests/associations-tests.bal (hunk 3 above)
Comment (kaneeldias) [reply ballerina/tests/associations-tests.bal]: Fixed in [3ac4df4](https://github.com/ballerina-platform/module-ballerina-persist/pull/155/commits/3ac4df4447e4c445dda487eb932865f44b7f670c)

File: ballerina/metadata_types.bal (hunk 2 above)
Comment (kaneeldias) [reply ballerina/metadata_types.bal]: Fixed in [3ac4df4](https://github.com/ballerina-platform/module-ballerina-persist/pull/155/commits/3ac4df4447e4c445dda487eb932865f44b7f670c)

File: ballerina/sql_client.bal (hunk 1 above)
Comment (kaneeldias) [reply ballerina/sql_client.bal]: Fixed in [3ac4df4](https://github.com/ballerina-platform/module-ballerina-persist/pull/155/commits/3ac4df4447e4c445dda487eb932865f44b7f670c)

**Pull request merged**